### PR TITLE
refactor(Scripts/SSC): The Lurker Below boss script updated

### PR DESCRIPTION
**Pull request by elthehablo**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

N.A.

## SOURCE:

The changes have been validated through:
N.A.

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [x] This pull request requires further testing and may have edge cases to be tested.

## Known Issues and TODO List:

- [x] sniffs

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp [hunk 1]
```diff
@@ -59,152 +61,154 @@ const Position positions[MAX_SUMMONS] =
...
-    struct boss_the_lurker_belowAI : public BossAI
+    void JustSummoned(Creature* summon) override
     {
-        boss_the_lurker_belowAI(Creature* creature) : BossAI(creature, DATA_THE_LURKER_BELOW) { }
+        summon->SetInCombatWithZone();
+        summons.Summon(summon);
+    }
 
-        void Reset() override
+    void DoAction(int32 param) override
```
Comment (Kitzunu) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp:91]: I see this as both action and param in the core. But I would prefer int32 action

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp [hunk 2]
```diff
@@ -59,152 +61,154 @@ const Position positions[MAX_SUMMONS] =
...
+    void Reset() override
     {
-        return GetSerpentShrineAI<boss_the_lurker_belowAI>(creature);
+        BossAI::Reset();
+        me->SetReactState(REACT_PASSIVE);
+        me->SetStandState(UNIT_STAND_STATE_SUBMERGED);
+        me->SetVisible(false);
+        me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+        // Reset summons
+        //summons.DespawnAll(); shouldn't be needed
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp:82]: bye bye

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp [hunk 3]
```diff
@@ -59,152 +61,154 @@ const Position positions[MAX_SUMMONS] =
...
 
-        void JustSummoned(Creature* summon) override
-        {
-            summon->SetInCombatWithZone();
-            summons.Summon(summon);
-        }
+    void JustDied(Unit* killer) override
+    {
+        BossAI::JustDied(killer);
+    }
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp:108]: Not needed

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp [hunk 4]
```diff
@@ -59,152 +61,154 @@ const Position positions[MAX_SUMMONS] =
...
-                            break;
-                        }
-            }
+        Unit* target = nullptr;
+        if (me->IsWithinMeleeRange(me->GetVictim()))
+            target = me->GetVictim();
+        else
+        {
+            ThreatContainer::StorageType const& t_list = me->GetThreatMgr().GetThreatList();
+            for (ThreatContainer::StorageType::const_iterator itr = t_list.begin(); itr != t_list.end(); ++itr)
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp:196]: Use range based for if possible

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**